distribution_metadata.c

```diff
@@ -419,28 +370,30 @@ PartitionType(Oid distributedTableId)
 bool
 IsDistributedTable(Oid tableId)
 {
+	Oid metadataNamespaceOid = get_namespace_oid("pgs_distribution_metadata", false);
+	Oid tableNamespaceOid = get_rel_namespace(tableId);
 	bool isDistributedTable = false;
-	RangeVar *heapRangeVar = NULL;
-	Relation heapRelation = NULL;
-	HeapScanDesc scanDesc = NULL;
-	const int scanKeyCount = 1;
-	ScanKeyData scanKey[scanKeyCount];
-	HeapTuple heapTuple = NULL;
-
-	heapRangeVar = makeRangeVar(METADATA_SCHEMA_NAME, PARTITION_TABLE_NAME, -1);
-	heapRelation = relation_openrv(heapRangeVar, AccessShareLock);
+	Oid argTypes[] = { OIDOID };
+	Datum argValues[] = { ObjectIdGetDatum(tableId) };
+	const int argCount = sizeof(argValues) / sizeof(argValues[0]);
+	int spiStatus = 0;
 
-	ScanKeyInit(&scanKey[0], ATTR_NUM_PARTITION_RELATION_ID, InvalidStrategy,
-				F_OIDEQ, ObjectIdGetDatum(tableId));
+	if (tableNamespaceOid == metadataNamespaceOid)
```

Comment:
> We definitely need a comment here explaining what is going on. Is there a better way to model this?

I'd rather use a comment instead of a separate function. It's effectively only one line… moving it out will hurt locality, and there's no need for the call outside of here.

By requiring callers to check first, we open ourselves up to an infinite loop if the caller fails to do so.

> Fundamentally, this check should only be needed for the partition table, is that correct?

Correct. Since its rows are the top-level objects in the graphs of distribution metadata for tables, putting the short-circuit here is sufficient.